packages/http/src/mocker/negotiator/__tests__/NegotiatorHelpers.spec.ts

```diff
@@ -738,189 +739,230 @@ describe('NegotiatorHelpers', () => {
-        assertLeft(negotiationResult, e => {
-          expect(e.message).toBe('The server cannot find the requested content');
+      it('should take the first content type', () => {
+        assertRight(negotiationResult, result => {
+          expect(result.mediaType).toBe('application/xml');
         });
       });
     });
+  });
+});
 
-    describe('given exampleKey not forced but dynamic forced', () => {
-      it('and httpContent has schema return that contents', () => {
-        const partialOptions = {
-          code: chance.integer({ min: 100, max: 599 }).toString(),
-          dynamic: true,
-        };
-        const httpContent: IMediaTypeContent = {
-          mediaType: chance.string(),
-          examples: [],
-          schema: { type: 'string' },
-          encodings: [],
-        };
+describe('negotiateByPartialOptionsAndHttpContent()', () => {
+  describe('given exampleKey forced', () => {
+    it('and example exists should return that example', () => {
+      const exampleKey = chance.string();
+      const partialOptions = {
+        code: chance.integer({ min: 100, max: 599 }).toString(),
+        exampleKey,
+        dynamic: chance.bool(),
+      };
+      const bodyExample: INodeExample = {
+        key: exampleKey,
+        value: '',
+      };
 
-        const actualOperationConfig = helpers.negotiateByPartialOptionsAndHttpContent(partialOptions, httpContent);
+      const httpContent: IMediaTypeContent = {
+        mediaType: chance.string(),
+        examples: [bodyExample],
+        encodings: [],
+      };
 
-        assertRight(actualOperationConfig, operationConfig => {
-          expect(operationConfig).toEqual({
-            code: partialOptions.code,
-            mediaType: httpContent.mediaType,
-            schema: httpContent.schema,
-          });
-        });
-      });
+      const actualOperationConfig = helpers.negotiateByPartialOptionsAndHttpContent(partialOptions, httpContent);
 
-      it('and httpContent has no schema throw error', () => {
-        const partialOptions = {
-          code: chance.integer({ min: 100, max: 599 }).toString(),
-          dynamic: true,
-        };
-        const httpContent: IMediaTypeContent = {
-          mediaType: chance.string(),
-          examples: [],
-          encodings: [],
-        };
+      const expectedConfig: Omit<IHttpNegotiationResult, 'headers'> = {
+        code: partialOptions.code,
+        mediaType: httpContent.mediaType,
+        bodyExample,
+      };
+
+      assertRight(actualOperationConfig, operationConfig => {
+        expect(operationConfig).toEqual(expectedConfig);
+      });
+    });
 
-        const negotiationResult = helpers.negotiateByPartialOptionsAndHttpContent(partialOptions, httpContent);
+    it('and example not exist should throw an error', () => {
+      const exampleKey = chance.string();
+      const partialOptions = {
+        code: chance.integer({ min: 100, max: 599 }).toString(),
```

Comment:
I'm wondering whether this is good to have non-deterministic test input. If somehow at some point it causes failure then it will be a nightmare to track it down.

I'm curious about your opinion on that?

Reply:
Yes, I'm ok fixing the status code, although there are other pieces using chance everywhere in this file — I do not know what the original intentions were.